packages/next/pages/_document.tsx

```diff
@@ -121,12 +121,49 @@ function getPreNextWorkerScripts(context: HtmlProps, props: OriginProps) {
           }}
         />
         {(scriptLoader.worker || []).map((file: ScriptProps, index: number) => {
-          const { strategy, ...scriptProps } = file
+          const {
+            strategy,
+            src,
+            children: scriptChildren,
+            dangerouslySetInnerHTML,
+            ...scriptProps
+          } = file
+
+          let srcProps: {
+            src?: string
+            dangerouslySetInnerHTML?: {
+              __html: string
+            }
+          } = {}
+
+          if (src) {
+            // Use external src if provided
+            srcProps.src = src
+          } else {
+            // Embed inline script if provided with dangerouslySetInnerHTML
+            if (dangerouslySetInnerHTML && dangerouslySetInnerHTML.__html) {
+              srcProps.dangerouslySetInnerHTML = {
+                __html: dangerouslySetInnerHTML.__html,
+              }
+            } else if (scriptChildren) {
+              // Embed inline script if provided with children
+              srcProps.dangerouslySetInnerHTML = {
+                __html:
+                  typeof scriptChildren === 'string'
+                    ? scriptChildren
+                    : Array.isArray(scriptChildren)
+                    ? scriptChildren.join('')
+                    : '',
+              }
+            }
```

Comment:
What about if none of these statements are true. Should there be a warning or error?

Reply:
Yeah I wasn't sure if I should add a warning here for when a user forgets to include a `src` or inline script. Probably safer to do that though.

Just added.